### PR TITLE
Simplify trackVerified(), expose RadarVerifiedLocationToken

### PR DESCRIPTION
**Pull request by nickpatrick**
- Remove `trackVerifiedToken()` and change `trackVerified()` to return `RadarVerifiedLocationToken`
- Remove `token` param from `startTrackingVerified()` and change `RadarVerifiedDelegate` to return `RadarVerifiedLocationToken`
- Make `token.expiresAt` override `interval` if needed
- Add `getVerifiedLocationToken()` which will return last token if not expired and request fresh token if needed

Other stuff:
- While making breaking changes, remove deprecated `autocomplete()` with `expandUnits` (already removed on Android)
- Random logging and comment cleanup: log format consistency, fix typos, remove unnecessary/obvious comments, lowercase one-line comments

### CODE REVIEW

File: RadarSDK/Radar.m [hunk 1]
```diff
@@ -759,28 +764,6 @@ + (void)searchGeofencesNear:(CLLocation *_Nullable)near
     }
 }
 
-+ (void)autocompleteQuery:(NSString *_Nonnull)query
```
Comment (david-goodfellow) [inline RadarSDK/Radar.m:762]: same

File: RadarSDK/RadarAPIClient.m [hunk 2]
```diff
@@ -473,8 +456,12 @@ - (void)trackWithLocation:(CLLocation *_Nonnull)location
                                 if (events.count) {
                                     [[RadarDelegateHolder sharedInstance] didReceiveEvents:events user:user];
                                 }
+                                
+                                if (token) {
```
Comment (david-goodfellow) [inline RadarSDK/RadarAPIClient.m:460]: by reading the code, wouldn't token always exist if user does?
```
id userObj = res[@"user"];
...
RadarUser *user = [[RadarUser alloc] initWithObject:userObj];
...
RadarVerifiedLocationToken *token = [[RadarVerifiedLocationToken alloc] initWithObject:res];
```

File: RadarSDK/Include/Radar.h [hunk 3]
```diff
@@ -874,27 +867,6 @@ Gets the device's current location, then searches for geofences near that locati
...
- @param near A location for the search.
- @param layers Optional layer filters.
- @param limit The max number of addresses to return. A number between 1 and 100.
- @param country An optional country filter. A string, the unique 2-letter country code.
- @param expandUnits Whether to expand units. Default behavior in other function signatures is false.
- @param completionHandler A completion handler.
-
- @see https://radar.com/documentation/api#autocomplete
- */
-+ (void)autocompleteQuery:(NSString *_Nonnull)query
```
[the remark this answers is not in the document]
Comment (lmeier) [reply RadarSDK/Include/Radar.h:890]: Left a note below, but I was imagining holding off on removal of this till `4.0.0`, but we can go ahead and do so

https://github.com/radarlabs/radar-sdk-ios/pull/328#discussion_r1585106542

File: RadarSDK/Include/Radar.h [hunk 4]
```diff
@@ -463,41 +464,33 @@ typedef void (^_Nonnull RadarLogConversionCompletionHandler)(RadarStatus status,
...
- Starts tracking the user's location with device integrity information for location verification use cases.
- 
- @param token A boolean indicating whether to return a JSON Web Token (JWT). If `true`, tokens are delivered to your `RadarVerifiedDelegate`. If `false`, location updates are delivered to your `RadarDelegate`.
- @param beacons A boolean indicating whether to range beacons.
- @param interval The interval in seconds between each location update. A number between 1 and 60.
- 
- @warning Note that you must configure SSL pinning before calling this method.
- */
-+ (void)startTrackingVerified:(BOOL)token interval:(NSTimeInterval)interval beacons:(BOOL)beacons NS_SWIFT_NAME(startTrackingVerified(token:interval:beacons:));
++ (void)getLastVerifiedLocationToken:(RadarTrackVerifiedCompletionHandler _Nullable)completionHandler NS_SWIFT_NAME(getLastVerifiedLocationToken(completionHandler:));
```
[the remark this answers is not in the document]
Comment (nickpatrick) [reply RadarSDK/Include/Radar.h:493]: Yes, will update to `getVerifiedLocationToken`

File: RadarSDK/Include/Radar.h [hunk 5]
```diff
@@ -874,27 +867,6 @@ Gets the device's current location, then searches for geofences near that locati
...
- @see https://radar.com/documentation/api#autocomplete
- */
-+ (void)autocompleteQuery:(NSString *_Nonnull)query
-                     near:(CLLocation *_Nullable)near
-                   layers:(NSArray<NSString *> *_Nullable)layers
-                    limit:(int)limit
-                  country:(NSString *_Nullable)country
-              expandUnits:(BOOL)expandUnits
-        completionHandler:(RadarGeocodeCompletionHandler)completionHandler NS_SWIFT_NAME(autocomplete(query:near:layers:limit:country:expandUnits:completionHandler:)) __deprecated;
-
```
[the remark this answers is not in the document]
Comment (nickpatrick) [reply RadarSDK/Include/Radar.h:897]: I can undo this removal

**Pull request closed without merging**